### ISSUE TITLE
data parallel training uses local init score for some objectives, but should use a global init score

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#4405]
## Description

Created from https://github.com/microsoft/LightGBM/pull/4332#issuecomment-853657414.

During distributed training, when a user-provided `init_score` is not given (the default), each worker determines an initial score to boost from based on its local data. These initial scores are then synced up by mean between all workers.

In situations where the distribution of the target is very different across different workers, this could lead to a lower-quality initial score, which might increase the number of boosting rounds it takes to fit to the training data.

#4332 addressed this for binary classification, but this problem is still present in other objectives. See https://github.com/microsoft/LightGBM/pull/4332#pullrequestreview-686980969.

## Environment info

LightGBM version or commit hash: https://github.com/microsoft/LightGBM/commit/0701a32da9fae7b2de8a01d702fa7d6abf36e836

Command(s) you used to install LightGBM: this affects any installation of LightGBM used for distributed training.

## Additional Comments

I've given this the label "dask" because it will affect training with `lightgbm.dask`, but it is not specific to this project's Dask interface. Any other distributed training option (https://lightgbm.readthedocs.io/en/latest/Parallel-Learning-Guide.html#integrations) will be affected by this as well.

**Comment by jameslamb** [comment]
@shiyu1994 please feel free to edit the description to add more details or fix anything I've written that is misleading or incorrect.

copying @imatiach-msft , as I think this issue is relevant for `mmlspark` as well.
